src/Lykke.Service.ClientDialogs.AzureRepositories/ClientDialog/ClientDialogsRepository.cs

```diff
@@ -0,0 +1,163 @@
+        {
+            _tableStorage = tableStorage;
+            _clientDialogIndex = clientDialogIndex;
+            _globalDialogIndex = globalDialogIndex;
+        }
+
+        public async Task<IClientDialog> AddDialogAsync(IClientDialog clientDialog)
+        {
+            var entity = ClientDialogEntity.Create(clientDialog);
+            await _tableStorage.InsertOrMergeAsync(entity);
+            return entity;
+        }
+
+        public async Task<IEnumerable<IClientDialog>> GetDialogsAsync()
+        {
+            return await _tableStorage.GetDataAsync();
+        }
+
+        public async Task<IClientDialog> GetDialogAsync(string dialogId)
+        {
+            return await _tableStorage.GetDataAsync(ClientDialogEntity.GeneratePartitionKey(),
+                ClientDialogEntity.GenerateRowKey(dialogId));
+        }
+
+        public Task AssignDialogToAllAsync(string dialogId)
+        {
+            var indexEntity = CreateGlobalDialogIndex(dialogId);
+            return _globalDialogIndex.InsertOrMergeAsync(indexEntity);
+        }
+
+        public Task UnAssignGlobalDialogAsync(string dialogId)
+        {
+            return _globalDialogIndex.DeleteIfExistAsync(ClientDialogEntity.GenerateGlobalDialogPartitionKey(), dialogId);
+        }
+
+        public async Task DeleteDialogAsync(string dialogId)
+        {
+            await _tableStorage.DeleteIfExistAsync(ClientDialogEntity.GeneratePartitionKey(), ClientDialogEntity.GenerateRowKey(dialogId));
+
+            var tasks = new List<Task>();
+            
+            var dialogIndexes = (await _clientDialogIndex.GetDataAsync(ClientDialogEntity.GenerateDialogIndex(dialogId))).ToList();
+
+            foreach (var index in dialogIndexes)
+            {
+                tasks.Add(_clientDialogIndex.DeleteIfExistAsync(index.RowKey, dialogId));
+                tasks.Add(_clientDialogIndex.DeleteIfExistAsync(ClientDialogEntity.GenerateDialogIndex(dialogId), index.RowKey));
+            }
+            
+            tasks.Add(UnAssignGlobalDialogAsync(dialogId));
+
+            await Task.WhenAll(tasks);
+        }
+
+        public Task SetDialogConditionTypeAsync(string dialogId, DialogConditionType? type)
+        {
+            return _tableStorage.ReplaceAsync(ClientDialogEntity.GeneratePartitionKey(),
+                ClientDialogEntity.GenerateRowKey(dialogId),
+                entity =>
+                {
+                    entity.ConditionType = type;
+                    return entity;
+                });
+        }
+
+        public async Task<IEnumerable<IClientDialog>> GetClientDialogsAsync(string clientId)
+        {
+            var indexesTask = _clientDialogIndex.GetDataAsync(clientId);
+            var globalIndexesTask = _globalDialogIndex.GetDataAsync(ClientDialogEntity.GenerateGlobalDialogPartitionKey());
+
+            await Task.WhenAll(indexesTask, globalIndexesTask);
+
+            var indexes = indexesTask.Result;
```

Comment:
never use .Result and Wait() - it is blocking

Reply:
it's already awaited, just grab the results, or its wrong?

Reply:
In that case no, just looks like a flag for me.

However Task.WhenAll can return results. Why not to use them and use .Result afterwards?